Sketch.java

```diff
@@ -1,36 +1,206 @@
+    for(int index = 0; index < circleX.length; index++){
+      circleX[index] = random(width);
+    }
+
+    // Array to store snowball random Y values
+    for(int index = 0; index < circleY.length; index++){
+      circleY[index] = random(height - 150);
+    }
   }
 
   /**
    * Called repeatedly, anything drawn to the screen goes here
    */
+
+  /**
+   * Draws snowball
+   * 
+   * @param blnPressDown boolean of down arrow key pressed
+   * @param blnPressUp boolean of up arrow key pressed
+   * @param intCircleYSpeed y speed of circle
+   * @param circleX x-coordinate of the ellipse
+   * @param circleY y-coordinate of the ellipse
+   * @param intPlayerX x-coordinate of user
+   * @param intPlayerY y-coordinate of user
+   * @param blnCharW boolean of W key pressed
+   * @param blnCharA boolean of A key pressed
+   * @param blnCharS boolean of S key pressed
+   * @param blnCharD boolean of D key pressed
+   */
+
   public void draw() {
-	  
-	// sample code, delete this stuff
-    stroke(128);
-    line(150, 25, 270, 350);  
+    background(130, 180, 240);
+
+    // Snowball speed based on user inputted arrow keys
+    if (blnPressDown && intCircleYSpeed < 20){
+      intCircleYSpeed ++;
+    }
+    else if (blnPressUp && intCircleYSpeed > 1){
+      intCircleYSpeed --;
+    }
+    
+    // Creates/show new snowballs on screen with circleX and circleY arrays on the condition of ballHideStatus[]
+    for (int index = 0; index < circleY.length; index++) {
+      fill(255);
+      if(!ballHideStatus[index]){
+        ellipse(circleX[index], circleY[index], 25, 25);
+        circleY[index] += intCircleYSpeed;
+        
+        if (circleY[index] > height) {
+          circleY[index] = 0;
+        }
+      }
+    }
 
-    stroke(255);
-    line(50, 125, 70, 50);  
+    // ballHideStatus[] tracks whether the user clicks on a snowball, and makes the snowball disappear
+    for(int index = 0; index < circleY.length; index++){
+      if(dist(mouseX, mouseY, (circleX[index]), circleY[index]) <= 12.5){
+        if(mousePressed){
+          ballHideStatus[index] = true;
+        }
+      }
+    }
+
+    // Player movement and WASD keys user input
+    fill(0, 0, 255);
+    ellipse(intPlayerX, intPlayerY, 30, 30);
+    if (blnCharW){
+      intPlayerY -= 5;
+    }
+    else if (blnCharA){
+      intPlayerX -= 5;
+    }
+    else if (blnCharS){
+      intPlayerY += 5;
+    }
+    else if (blnCharD){
+      intPlayerX += 5;
+    }
+
+    // Player edge collision detection
+    if (intPlayerX > width - 15){
+      intPlayerX = width - 15;
+    }
+    else if (intPlayerX < 15){
+      intPlayerX = 15;
+    }
+    else if (intPlayerY < 15){
+      intPlayerY = 15;
+    }
+    else if (intPlayerY > height - 15){
+      intPlayerY = height - 15;
+    }
+
+    // Deduct player lives when the collide with a snowball
+    for(int index = 0; index < circleY.length; index++){
+      if(dist(intPlayerX, intPlayerY, (circleX[index]), circleY[index]) <= 30){
```

Comment:
collision should also include consideration of `ballHideStatus` otherwise you collided with invisible balls.